Lab7/src/main/java/agh/ics/oop/WorldMap.java

```diff
@@ -0,0 +1,46 @@
+package agh.ics.oop;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class WorldMap implements IWorldMap{
+
+
+    protected Map<Vector2d, Animal> animals = new HashMap<>();
+
+    @Override
+    public Map<Vector2d, Animal> getMapOfAnimals() {
+        return Map.copyOf(this.animals);
+    }
+
+    @Override
+    public boolean place(Animal animal) {
```

Comment:
skoro rzucamy wyjątkiem to nie zwrócimy tu nigdy false -> ta metoda nie powinna już zwracać boolean